presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java

```diff
@@ -362,6 +377,17 @@ else if (resourceManager) {
             eventListenerManager = ((TestingEventListenerManager) injector.getInstance(EventListenerManager.class));
             clusterStateProvider = injector.getInstance(ResourceManagerClusterStateProvider.class);
         }
+        else if (catalogServer) {
+            dispatchManager = null;
+            queryManager = injector.getInstance(QueryManager.class);
```

Comment:
Why is `QueryManager` needed?

Reply:
I did that since we assigned its query manager in the CatalogServerModule. Since its NoOp anyways I can probably put it to null and will have no effect